pkg/api/kptlib/clone.go

```diff
@@ -0,0 +1,52 @@
+// Copyright 2022 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package kptlib
+
+import (
+	"fmt"
+	"strings"
+
+	internalpkg "github.com/GoogleContainerTools/kpt/internal/pkg"
+	kptfilev1 "github.com/GoogleContainerTools/kpt/pkg/api/kptfile/v1"
+	"sigs.k8s.io/kustomize/kyaml/yaml"
+)
+
+// TODO: Accept a virtual filesystem or other package abstraction
+func UpdateUpstreamFromGit(kptfileContents string, name string, lock kptfilev1.GitLock) (string, error) {
```

Comment:
Looks like a better place for this method will be`pkg/kptfile/kptfileutil` pkg.

Reply:
Thanks! I moved the file inside Porch for now as we finalize the locations for interfaces.